pkg/api/scylla/validation/scyllaoperatorconfig_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2023 ScyllaDB.
+
+package validation_test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/api/scylla/validation"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateScyllaOperatorConfig(t *testing.T) {
+	t.Parallel()
+
+	tt := []struct {
+		name                 string
+		ScyllaOperatorConfig *scyllav1alpha1.ScyllaOperatorConfig
+		expectedErrorList    field.ErrorList
+		expectedErrorString  string
+	}{
+		{
+			name:                 "empty config is valid",
+			ScyllaOperatorConfig: &scyllav1alpha1.ScyllaOperatorConfig{},
+			expectedErrorList:    field.ErrorList{},
+			expectedErrorString:  "",
+		},
+		{
+			name: "UnsupportedGrafanaImageOverride can't be empty",
+			ScyllaOperatorConfig: &scyllav1alpha1.ScyllaOperatorConfig{
+				Spec: scyllav1alpha1.ScyllaOperatorConfigSpec{
+					UnsupportedGrafanaImageOverride: pointer.Ptr(""),
+				},
+			},
+			expectedErrorList: field.ErrorList{
+				&field.Error{
+					Type:     field.ErrorTypeRequired,
+					Field:    "spec.unsupportedGrafanaImageOverride",
+					BadValue: "",
+					Detail:   "image reference can't be empty",
+				},
+			},
+			expectedErrorString: `spec.unsupportedGrafanaImageOverride: Required value: image reference can't be empty`,
+		},
+		{
+			name: "UnsupportedGrafanaImageOverride can't contain only spaces",
+			ScyllaOperatorConfig: &scyllav1alpha1.ScyllaOperatorConfig{
+				Spec: scyllav1alpha1.ScyllaOperatorConfigSpec{
+					UnsupportedGrafanaImageOverride: pointer.Ptr(" "),
+				},
+			},
+			expectedErrorList: field.ErrorList{
+				&field.Error{
+					Type:     field.ErrorTypeInvalid,
+					Field:    "spec.unsupportedGrafanaImageOverride",
+					BadValue: " ",
+					Detail:   "unable to parse image: invalid reference format",
+				},
+			},
+			expectedErrorString: `spec.unsupportedGrafanaImageOverride: Invalid value: " ": unable to parse image: invalid reference format`,
+		},
+		{
+			name: "UnsupportedGrafanaImageOverride accepts valid image",
+			ScyllaOperatorConfig: &scyllav1alpha1.ScyllaOperatorConfig{
+				Spec: scyllav1alpha1.ScyllaOperatorConfigSpec{
+					UnsupportedGrafanaImageOverride: pointer.Ptr("docker.io/scylladb/scylla-operator:latest"),
```

Comment:
nit: add a test for non-tagged image reference as well?

Reply:
added